tdrs-backend/tdpservice/search_indexes/documents/ssp.py

```diff
@@ -161,6 +186,18 @@ class Django:
 class SSP_M3DataSubmissionDocument(Document):
     """Elastic search model mapping for a parsed SSP M3 data file."""
 
+    datafile = fields.ObjectField(properties={
+                      'pk': fields.IntegerField(),
+                      'created_at': fields.DateField(),
+                      'version': fields.IntegerField(),
+                      'quarter': fields.TextField()
+                  })
+
+    def get_instances_from_related(self, related_instance):
+        """Return correct instance."""
+        if isinstance(related_instance, DataFile):
+            return related_instance
```

Comment:
Third occurrence of this function. Possible to refactor to pull out of class definitions? Seems like a silly use-case for a parent class but it's borderline.

Reply:
I can pull this out into a base class and let the documents inherit from it. But it has to be a member of the class and cant be a standalone function.

Reply:
@andrew-jameson let me know what you think of the base class